Mathlib/Data/Matrix/Basic.lean

```diff
@@ -2731,22 +2799,25 @@ variable [Fintype n] [NonAssocSemiring α] [NonAssocSemiring β]
 
 theorem map_matrix_mul (M : Matrix m n α) (N : Matrix n o α) (i : m) (j : o) (f : α →+* β) :
     f (Matrix.mul M N i j) = Matrix.mul (M.map f) (N.map f) i j := by
-  simp [Matrix.mul_apply, RingHom.map_sum]
+  simp [Matrix.mul_apply, map_sum]
 #align ring_hom.map_matrix_mul RingHom.map_matrix_mul
 
 theorem map_dotProduct [NonAssocSemiring R] [NonAssocSemiring S] (f : R →+* S) (v w : n → R) :
-    f (v ⬝ᵥ w) = f ∘ v ⬝ᵥ f ∘ w := by simp only [Matrix.dotProduct, f.map_sum, f.map_mul]
+    f (v ⬝ᵥ w) = f ∘ v ⬝ᵥ f ∘ w := by
+  simp only [Matrix.dotProduct, f.map_sum, f.map_mul]
+  simp_rw [Function.comp_apply (f := f)] -- Porting note: (simp) this line wasnt needed before
 #align ring_hom.map_dot_product RingHom.map_dotProduct
 
 theorem map_vecMul [NonAssocSemiring R] [NonAssocSemiring S] (f : R →+* S) (M : Matrix n m R)
     (v : n → R) (i : m) : f (M.vecMul v i) = (M.map f).vecMul (f ∘ v) i := by
   simp only [Matrix.vecMul, Matrix.map_apply, RingHom.map_dotProduct]
+  rfl -- Porting note: (rfl) in mathlib3 this `rfl` wasn't needed
 #align ring_hom.map_vec_mul RingHom.map_vecMul
 
 theorem map_mulVec [NonAssocSemiring R] [NonAssocSemiring S] (f : R →+* S) (M : Matrix m n R)
     (v : n → R) (i : m) : f (M.mulVec v i) = (M.map f).mulVec (f ∘ v) i := by
   simp only [Matrix.mulVec, Matrix.map_apply, RingHom.map_dotProduct]
+  rfl -- Porting note: (rfl) in mathlib3 this `rfl` wasn't needed
```

Comment:
What's left here?

Reply:
I don't understand why we need that `rfl` after `simp`? (I also haven't looked into it). Is that just a known regression of lean4-`simp` vs the lean3-version, or is there something to fix here?

Reply:
Another option would be to have a `simp`-lemma:
```
@[simp]
theorem Function.comp_fun (f : α → β) (g : γ → α) : f ∘ (fun i => g i) = fun i => f (g i) := rfl
```

It might be because `Function.comp` had the `reducible` attribute in lean3-core, but does not in lean4-core.

Reply:
Adding `Function.comp` to the simp call is the usual approach I think

Reply:
I see, Ill do that!